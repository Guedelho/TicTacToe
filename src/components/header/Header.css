.Header {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  height: auto; }
  .Header.blue h1 {
    color: #007bff; }
  .Header.green h1 {
    color: #28a745; }
  .Header.white h1 {
    color: #fff; }
  .Header h1 {
    text-align: center;
    margin: 0; }
    @media (min-width: 480px) {
      .Header h1 {
        margin: 20px; } }
