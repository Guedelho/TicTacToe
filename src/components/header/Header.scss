$black: #343a40;
$green: #28a745;
$blue: #007bff;
$red: #dc3545;

.Header {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  height: auto;
  &.blue {
    h1{
      color: $blue;
    }
  }
  &.green {
    h1{
      color: $green;
    }
  }
  &.white {
    h1{
      color: #fff;
    }
  }
}

h1 {
  text-align: center;
}