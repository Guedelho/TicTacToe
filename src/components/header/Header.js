import React from 'react';

import './Header.css';

const Header = () => (
  <header className="Header">
    <h1>Jogo da velha da Jazida</h1>
  </header>
);

export default Header;
