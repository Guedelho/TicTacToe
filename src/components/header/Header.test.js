import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

const props = {
  result: 0,
  name: "string",
  playerName: "string",
  playerColor: "string",
}

it('renders without crashing', () => {
  expect(shallow(<Header {...props}/>)).toMatchSnapshot();
});