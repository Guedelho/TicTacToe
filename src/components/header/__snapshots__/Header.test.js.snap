// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    name="string"
    playerColor="string"
    playerName="string"
    result={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        string
        , é a sua vez!
</h1>,
      "className": "Header string false",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "string",
          ", é a sua vez!",
        ],
      },
      "ref": null,
      "rendered": Array [
        "string",
        ", é a sua vez!",
      ],
      "type": "h1",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          string
          , é a sua vez!
</h1>,
        "className": "Header string false",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "string",
            ", é a sua vez!",
          ],
        },
        "ref": null,
        "rendered": Array [
          "string",
          ", é a sua vez!",
        ],
        "type": "h1",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
